cmd/SplitGrantsGovXMLDB/handler.go

```diff
@@ -157,34 +149,44 @@ func readOpportunities(ctx context.Context, r io.Reader, ch chan<- opportunity)
 			return err
 		}
 
-		// When reading the start of a new element, check if it is a grant opportunity
-		se, ok := token.(xml.StartElement)
-		if ok && se.Name.Local == GRANT_OPPORTUNITY_XML_NAME {
-			var opportunity opportunity
-			if err := d.DecodeElement(&opportunity, &se); err != nil {
-				level.Error(logger).Log("msg", "Error decoding XML token", "error", err)
+		// When reading the start of a new element, check if it is a grant opportunity or forecast
+		if se, ok := token.(xml.StartElement); ok {
+			var err error
+			if se.Name.Local == GRANT_OPPORTUNITY_XML_NAME {
+				var o opportunity
+				if err = d.DecodeElement(&o, &se); err == nil {
+					ch <- &o
+				}
+			} else if se.Name.Local == GRANT_FORECAST_XML_NAME {
+				var f forecast
+				if err = d.DecodeElement(&f, &se); err == nil {
+					ch <- &f
+				}
+			}
```

Comment:
I wonder if we should comment this block or otherwise prevent sending the decoded `forecast` to the channel until the downstream DynamoDB persistence step is ready to handle `OpportunityForecastDetail_1_0.xml` object events. As of today, there are just over 600 forecast records in the extract; I expect this to be a similar number once these changes hit Production. If we roll this out as-is, we'll either need to bulk-delete the objects in S3 or otherwise manually trigger the `PersistGrantsGovXMLDB` step for each fetched object. While I don't think saving the forecast records to S3 ahead of downstream changes would cause any issues in the pipeline, it means that we'd miss out on being able to pull in the bulk of currently-forecasted grants.

Perhaps we should add a feature flag (env var) here that would allow us to still observe that things are working as-expected on Staging, which we could control in Terraform. 

Thoughts?

Reply:
Ah, I see... once this runs and the forecasted XML snippets are pushed to S3, they'll be considered extant unless they've been updated for subsequent runs. So when we eventually update the downstream tasks to handle forecasted grants, we won't automatically get the full set of them in the DB. Have I got the concern? 

I think this concern points to there being some need for a utility to rebuild the database, which would be useful in this situation and likely in others. (And possibly just a good occasional vacuuming type of task to run periodically.) That's out of scope here; I'm just mentioning it so it's in our heads for the future... 

For this PR, it sounds like a flag is the way to go. I'll get on implementing that!

Reply:
> I think this concern points to there being some need for a utility to rebuild the database, which would be useful in this situation and likely in others.

There are a few, let's say "scrappy" ways of doing this today – some equivalent of
```cli
aws s3 ls --recursive s3://the-bucket/path | grep 'pattern' | xargs aws lambda invoke ...
```

I've got some scratch work laying around to turn the contents of `cli/` into a more robust maintenance tool for these types of things, but point taken that this should be formalized – good call.